src/OpenFeature.Contrib.Providers.Flagsmith/FlagsmithProvider.cs

```diff
@@ -0,0 +1,167 @@
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        public FlagsmithProvider(IFlagsmithConfiguration options)
+        {
+            _flagsmithClient = new FlagsmithClient(options);
+        }
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        /// <param name="httpClient">Http client that will be used for flagsmith requests. You also can use it to register <see cref="FeatureProvider"/> as Typed HttpClient with <see cref="FeatureProvider"> as abstraction</see></param>
+        public FlagsmithProvider(IFlagsmithConfiguration options, HttpClient httpClient)
+        {
+            _flagsmithClient = new FlagsmithClient(options, httpClient);
+        }
+
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="flagsmithClient">Precreated Flagsmith client. You can just use <see cref="FlagsmithClient"/> class.</param>
+        public FlagsmithProvider(IFlagsmithClient flagsmithClient)
+        {
+            _flagsmithClient = flagsmithClient;
+        }
+
+        private Task<IFlags> GetFlags(EvaluationContext ctx)
+        {
+            var key = ctx?.GetValue(TargetingKey)?.AsString;
+            return string.IsNullOrEmpty(key)
+                ? _flagsmithClient.GetEnvironmentFlags()
+                : _flagsmithClient.GetIdentityFlags(key, ctx.AsDictionary().Select(x => new Trait(x.Key, x.Value.AsObject) as ITrait).ToList());
+        }
+
+        private async Task<ResolutionDetails<T>> ResolveValue<T>(string flagKey, T defaultValue, TryParseDelegate<T> tryParse, EvaluationContext context)
+        {
+
+            var flags = await GetFlags(context);
+            var isFlagEnabled = await flags.IsFeatureEnabled(flagKey);
+            if (!isFlagEnabled)
+            {
+                return new ResolutionDetails<T>(flagKey, defaultValue, reason: Reason.Disabled);
+            }
+
+            var stringValue = await flags.GetFeatureValue(flagKey);
+
+            if (tryParse(stringValue, out var parsedValue))
+            {
+                return new ResolutionDetails<T>(flagKey, parsedValue);
+            }
+            throw new TypeMismatchException("Failed to parse value in the expected type");
+
+        }
+
+
+        /// <inheritdoc/>
+        public override Metadata GetMetadata() => Metadata;
+
+        /// <inheritdoc/>
+
+        public override Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+            => ResolveValue(flagKey, defaultValue, bool.TryParse, context);
```

Comment:
Because of the fact that Flagsmith flags can have an enabled state, and a value, in the other Flagsmith providers we use an additional bit of config to allow users of the provider to state if they want to use the value or the state when resolving boolean values. See an example in the Java provider [here](https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProvider.java#L53-L56). 

Would it be possible to do something similar here please?

Reply:
sure. i'll try)

Reply:
done!